### PR TITLE
Update status

### PR DESCRIPTION
**Pull request by hedita**

### CODE REVIEW

File: assets/js/index.js [hunk 1]
```diff
@@ -85,15 +100,17 @@ function formatDate(date) {
...
+    .map(({ text, createdAt, id: taskId, isDone }) =>
+    `<li id="${taskId}" class="todo-item">
+    <input class="checkbox" type="checkbox" ${isChecked(isDone)}/>
     <p class="todo"  title="${formatDate(createdAt)}">${text}</p>
     <i class="fa-solid fa-pen-to-square edite-icon"></i>
     <i class="fa-solid fa-xmark remove-icon"></i>
     </li>`).join("");
   bindDeleteEvent();
+  bindUpdateEvent();
+
```
Comment (takrami) [inline assets/js/index.js]: extra line.

File: assets/js/index.js [hunk 2]
```diff
@@ -42,6 +42,21 @@ async function deleteTodo(taskId) {
...
 }
 
+async function updateStatus(taskId, isDone) {
+  try {
+    await fetch(`${url}/todos/${taskId}`, {
+      method: "PATCH",
+      body: JSON.stringify({ isDone }),
+      headers: {
+        "Content-Type": "application/json",
+      }
```
Comment (takrami) [inline assets/js/index.js]: This part of the code seems to be duplicated. Could you please move this to the global scope and re-use it? Something like `requestDefaultHeaders`.

File: assets/js/index.js [hunk 3]
```diff
@@ -133,4 +150,33 @@ function bindDeleteEvent() {
...
+  const checkboxs = Array.from(document.getElementsByClassName("checkbox"));
+  checkboxs.forEach(checkbox => {
+    checkbox.addEventListener("change", function (event) {
+      updateStatus(event.target.parentNode.id, this.checked);
+    })
+  });
+}
+
+function isChecked(isDone) {
+  if (isDone == true) {
```
Comment (takrami) [inline assets/js/index.js]: Why not `===`?

File: assets/js/index.js [hunk 4]
```diff
@@ -133,4 +150,33 @@ function bindDeleteEvent() {
...
+  }
+}
+
+function sortTasksByTime(tasks) {
+  return tasks.sort(
+    (a, b) => new Date(b.createdAt) - new Date(a.createdAt)
+  )
+}
+
+function sortTasksByIsDone(tasks) {
```
Comment (takrami) [inline assets/js/index.js]:
```suggestion
function sortTasksByStatus(tasks) {
```

File: assets/js/index.js (hunk 1 above)
Comment (hedita) [reply assets/js/index.js]: Done.

File: assets/js/index.js (hunk 3 above)
Comment (hedita) [reply assets/js/index.js]: Done.

File: assets/js/index.js (hunk 1 above)
Comment (hedita) [reply assets/js/index.js]: I think that was not extra.

File: assets/js/index.js (hunk 2 above)
Comment (hedita) [reply assets/js/index.js]: Done.

**Pull request merged**